### DISCUSSION TITLE
Is there a way to deny pushes but still allow merging PRs?

### DISCUSSION THREAD

**Discussion by aaronfranke** [community/community#23713] [New to GitHub]
<p>Occasionally someone makes a mistake by pushing something they shouldn’t, so it would be nice to deny pushing so that the only changes done to a repository are through merging PRs. I know that you can remove write access, but this also prevents merging PRs. Is it possible to only prevent pushes on GitHub?</p>

**Comment by francisfuzz** [comment, marked as answer]
<p><a class="mention" href="/aaronfranke">@aaronfranke</a> - Great question! Have you considered leveraging <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/about-protected-branches">protected branches</a> for your use case?</p>
<p>Protected branches ensure that collaborators on your repository cannot make irrevocable changes to branches. Enabling protected branches also allows you to enable other optional checks and requirements, like required status checks and required reviews. Our team wrote a guide for <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/defining-the-mergeability-of-pull-requests">defining the mergeability of pull requests</a> that may be worth checking out.</p>
<p>Once you have protected branches setup on your repository, those with <code>write</code> access will still be able to push to branches that are not protected, but any pushes from them attempting to update the protected branch(es) will be rejected.</p>
<p>If you’re just getting started with protected branches, one of my colleagues published a guide covering <a href="https://github.community/t/best-practices-for-protected-branches/10204">best practices for protected branches</a> that could be useful.</p>
<p>Does that help with what you’re looking to do? :thought_balloon:</p>
